test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -183,12 +185,26 @@ object TestAspect {
   /**
    * An aspect that retries failed tests according to a schedule.
    */
-  def retry[R0, E0](schedule: ZSchedule[R0, E0, Any]): TestAspect[Nothing, R0 with Clock, Nothing, E0] =
-    new TestAspect.PerTest[Nothing, R0 with Clock, Nothing, E0] {
-      def perTest[R >: Nothing <: R0 with Clock, E >: Nothing <: E0](
+  def retry[R0, E0](schedule: ZSchedule[R0, E0, Any]): TestAspect[Nothing, R0 with Live[Clock], Nothing, E0] =
+    new TestAspect.PerTest[Nothing, R0 with Live[Clock], Nothing, E0] {
+      def perTest[R >: Nothing <: R0 with Live[Clock], E >: Nothing <: E0](
         test: ZIO[R, E, TestResult]
-      ): ZIO[R, E, TestResult] =
-        test.retry(schedule: ZSchedule[R0, E0, Any])
+      ): ZIO[R, E, TestResult] = {
+        def loop(state: schedule.State): ZIO[R with Live[Clock], E, TestResult] =
+          test.foldM(
+            err =>
+              schedule
+                .update(err, state)
+                .flatMap(
+                  decision =>
+                    if (decision.cont) Live.live(clock.sleep(decision.delay)) *> loop(decision.state)
+                    else ZIO.fail(err)
+                ),
+            succ => ZIO.succeed(succ)
+          )
+
+        schedule.initial.flatMap(loop)
```

Comment:
How about getting the live clock with `access`, then `provide` it, so that we reuse the implementation of `retry`?

Reply:
I tried that but was struggling to make it work. The issue is that the schedule also has a `R` parameter other than `Clock` but as far as I am aware there isn't a method like `provide` to eliminate that requirement. So between not being able to eliminate the `R` requirements by providing them and the limitations on `provideSome` this was the best I could come up with.

Reply:
Oh yeah I see, there's no way to rebuild the R in `provideSome`...